app/models/bid.rb

```diff
@@ -11,8 +11,11 @@ class Bid < ApplicationRecord
 
   scope :this_month, -> {where(created_at: (Time.now.beginning_of_month..Time.zone.now))}
 
-  def toggle_accept(status)
-    update(accepted: !status)
+  def toggle_accept!(status)
+  	self.accepted = !status
+  	self.job.agent_id = accepted ? bidder.id : nil
+  	self.save
+  	self.job.save
```

Comment:
Hmm.That didn't work. The Bid would be updated but the agent_id in Job would not be updated at all.